File: leetcode/java/53.maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        if (nums.length == 1) return nums[0];
        int n = 0;
        int res = nums[0];
        for (int i = 0; i < nums.length; i++) {
            n += nums[i];
            res = Math.max(res, n);
            if (n < 0) {
                n = 0;
            }

        }
        return res;
    }
}

File: leetcode/cpp/53.maximum-subarray.cpp
#include "oj_header.h"
class Solution
{
public:
    int maxSubArray(vector<int> &nums)
    {
        if (nums.size() == 0)
            return 0;
        if (nums.size() == 1)
            return nums[0];
        int m = nums[0];
        int prev = m;
        for (int i = 1; i < nums.size(); ++i)
        {
            if (prev < 0)
                prev = nums[i];
            else
                prev += nums[i];
            m = max<int>(prev, m);
        }

        return m;
    }
};
